### PR TITLE
[WIP] Migrate Chat Session (V0) to Conversation (V2)

### PR DESCRIPTION
**Pull request by lasryaric**
This code has been tested only on a few local conversations for now.
From experience, I would highly recommend to run that on a dump of our database before ever running it in production.

LMK what you think @spolu .

### CODE REVIEW

**Comment by lasryaric** [comment]
Just realizing I forgot to create the `ConversationParticipant` object. Doing it now.

File: front/migrations/20230919_chat_v0_v2.ts [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+              parentId: previousMessage?.id || null,
+              userMessageId: newUserMessage?.id || null,
+              agentMessageId: newAgentMessage?.id || null,
+              conversationId: newConversation.id,
+            },
+            {
+              transaction: t,
+            }
+          )) as [Message, boolean | null];
+          previousMessage = newMessage;
```
Comment (lasryaric) [inline front/migrations/20230919_chat_v0_v2.ts:271]: previous message could be a retrieval here, not sure what we want to do.

File: front/migrations/20230919_chat_v0_v2.ts [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+              {
+                id:
+                  existingMessage.length > 0
+                    ? existingMessage[0].id
+                    : undefined,
+                createdAt: oldMessage.createdAt,
+                updatedAt: oldMessage.updatedAt,
+                content: oldMessage.message || "",
+                userContextUsername: user?.username || "",
+                userContextTimezone: "Europe/Paris", // Looks like we don't have any timezone in the DB today. TBD at code review time.
```
Comment (spolu) [inline front/migrations/20230919_chat_v0_v2.ts:133]: Yes LGTM

File: front/migrations/20230919_chat_v0_v2.ts [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+                userContextEmail: user?.email || null,
+                userContextFullName: user?.name || null,
+                userContextProfilePictureUrl: null, // looks like we don't store the user profile pictures today. TBD at code review time.
```
Comment (spolu) [inline front/migrations/20230919_chat_v0_v2.ts:136]: yes null

**Review by spolu — COMMENTED** [review]
Cool! left a few comments :+1:

File: front/migrations/20230919_chat_v0_v2.ts [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+                await getRetrievalForOldMessage(oldMessage);
+              if (retrievalMessage && !newAgentMessage.agentRetrievalActionId) {
+                // if we already have a agentRetrievalActionId, we don't re-migrate the retrieval as they are hard to dedup
+                const agentRetrievalAction = await AgentRetrievalAction.upsert(
+                  {
+                    id: newAgentMessage.agentRetrievalActionId || undefined,
+                    createdAt: retrievalMessage.createdAt,
+                    updatedAt: retrievalMessage.updatedAt,
+                    topK: 16,
+                    retrievalConfigurationId:
```
[the remark this answers is not in the document]
Comment (lasryaric) [reply front/migrations/20230919_chat_v0_v2.ts:204]: done

File: front/migrations/20230919_chat_v0_v2.ts (hunk 1 above)
Comment (lasryaric) [reply front/migrations/20230919_chat_v0_v2.ts]: Make sense.

**Comment by spolu** [comment]
This will need rebase on top of https://github.com/dust-tt/dust/pull/1598 simple addition of a `version: 0`

File: front/migrations/20230919_chat_v0_v2.ts [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+        }),
+        retrievalMessage: oldMessages[i - 1],
+      };
+    }
+  }
+
+  return { chatRetrievedDocuments: [], retrievalMessage: null };
+}
+
+async function main() {
```
[the remark this answers is not in the document]
Comment (lasryaric) [reply front/migrations/20230919_chat_v0_v2.ts:348]: Done.

**Review by lasryaric — COMMENTED** [review]
Addressed your comments @spolu.

**Comment by lasryaric** [comment]
>This will need rebase on top of https://github.com/dust-tt/dust/pull/1598 simple addition of a version: 0

Am I supposed to read/write this version field?

**Comment by lasryaric** [comment]
>This will need rebase on top of https://github.com/dust-tt/dust/pull/1598 simple addition of a version: 0

>Am I supposed to read/write this version field?

Got it.

**Review by spolu — DISMISSED** [review]
LGTM

Let's run it on our workspace?

**Pull request merged**